File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/fs/VfsRootAygor.java
/**
 * @file
 * @brief Implementation of VfsRoot over http://hvsc.c64.org collection
 * @author vitamin.caig@gmail.com
 */

package app.zxtune.fs;

import android.content.Context;
import android.net.Uri;

import androidx.annotation.Nullable;

import app.zxtune.R;
import app.zxtune.fs.aygor.Path;
import app.zxtune.fs.cache.CacheDir;
import app.zxtune.fs.http.MultisourceHttpProvider;
import app.zxtune.fs.httpdir.Catalog;
import app.zxtune.fs.httpdir.HttpRootBase;

@Icon(R.drawable.ic_browser_vfs_aygor)
final class VfsRootAygor extends HttpRootBase implements VfsRoot {

  private final Context context;

  VfsRootAygor(VfsObject parent, Context context, MultisourceHttpProvider http, CacheDir cache) {
    super(parent, Catalog.create(context, http, cache, "aygor"), Path.create());
    this.context = context;
  }

  @Override
  public String getName() {
    return context.getString(R.string.vfs_aygor_root_name);
  }

  @Override
  public String getDescription() {
    return context.getString(R.string.vfs_aygor_root_description);
  }

  @Override
  @Nullable
  public VfsObject resolve(Uri uri) {
    return resolve(Path.parse(uri));
  }
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/fs/VfsRootAygor.kt
/**
 * @file
 * @brief Implementation of VfsRoot over http://hvsc.c64.org collection
 * @author vitamin.caig@gmail.com
 */
package app.zxtune.fs

import android.content.Context
import android.net.Uri
import app.zxtune.R
import app.zxtune.fs.aygor.Path
import app.zxtune.fs.http.MultisourceHttpProvider
import app.zxtune.fs.httpdir.Catalog
import app.zxtune.fs.httpdir.HttpRootBase

internal class VfsRootAygor(
    parent: VfsObject,
    private val context: Context,
    http: MultisourceHttpProvider
) : HttpRootBase(parent, Catalog.create(context, http, "aygor"), Path.create()), VfsRoot {

    override val name: String
        get() = context.getString(R.string.vfs_aygor_root_name)
    override val description: String
        get() = context.getString(R.string.vfs_aygor_root_description)

    override fun getExtension(id: String) = when (id) {
        VfsExtensions.ICON -> R.drawable.ic_browser_vfs_aygor
        else -> super.getExtension(id)
    }

    override fun resolve(uri: Uri) = resolve(Path.parse(uri))
}
